pyanaconda/modules/common/structures/timezone.py

```diff
@@ -82,3 +82,124 @@ def options(self) -> List[Str]:
     @options.setter
     def options(self, value):
         self._options = value
+
+
+class GeolocationResult(DBusData):
+    """Encapsulates the result from geolocation lookup."""
+
+    def __init__(self):
+        self._status = ""
+        self._territory_code = ""
+        self._timezone = ""
+        self._timezone_source = ""
+        self._public_ip_address = ""
+        self._city = ""
+
+    @classmethod
+    def from_values(cls, status="", territory_code="", timezone="", timezone_source="unknown",
+                    public_ip_address="", city=""):
+        """
+        :param str status: status of the result, see below for possible states/values
+        :param str territory_code: the territory code from GeoIP lookup
+        :param str timezone: the time zone from GeoIP lookup
+        :param str timezone_source: specifies source of the time zone string
+        :param str public_ip_address: current public IP address
+        :param str city: current city
+        """
+        instance = cls()
+        instance.status = status
+        instance.territory_code = territory_code
+        instance.timezone = timezone
+        instance.timezone_source = timezone_source
+        instance.public_ip_address = public_ip_address
+        instance.city = city
+
+    @property
+    def status(self) -> Str:
+        """Status of the geolocation system and result
+
+        Can be one of:
+            NOT_STARTED
+            RUNNING
+            DONE
```

Comment:
What about just:

```
IDLE       The geolocation doesn't run.
DETECTING  The geolocation is running.
```

Reply:
That does not tell me if there is a result or not, though. If I start to rely on presence of data to detect there is something, I could just drop the status too.

Reply:
Based on the code, we don't need to know that. We just need to know if it is worth it to give the geolocation more time.